sycl/source/detail/event_impl.cpp

```diff
@@ -442,20 +453,47 @@ void event_impl::cleanDepEventsThroughOneLevel() {
 }
 
 void event_impl::setSubmissionTime() {
-  if (!MIsProfilingEnabled || MLimitedProfiling)
+  if (!MIsProfilingEnabled)
     return;
-  if (QueueImplPtr Queue = MQueue.lock()) {
-    try {
-      MSubmitTime = Queue->getDeviceImplPtr()->getCurrentDeviceTime();
-    } catch (feature_not_supported &e) {
-      throw sycl::exception(
-          make_error_code(errc::profiling),
-          std::string("Unable to get command group submission time: ") +
-              e.what());
+  if (!MFallbackProfiling) {
+    if (QueueImplPtr Queue = MQueue.lock()) {
+      try {
+        MSubmitTime = Queue->getDeviceImplPtr()->getCurrentDeviceTime();
+      } catch (feature_not_supported &e) {
+        throw sycl::exception(
+            make_error_code(errc::profiling),
+            std::string("Unable to get command group submission time: ") +
+                e.what());
+      }
     }
+  } else {
+    // 1) This code is in order to support profiling for OpenCL version < 2.1
+    // that API clGetDeviceAndHostTimer is not availalbe to obrain the precise
+    // host and device submit time. The host timestamp in sycl runtime will be
+    // return for the info::event_profiling::command_submit
+    // host timestamp to to return for calling <command_submit>
+    // this is the base time in host side to support fallback profiling
+    // for OpenCL version < 2.1, see MFallbackProfiling
+    // in fact, this fuction is called after enqueuing is processed
+    // successfully in piEnqueueKernelLaunch()
+    // TO DO: submit time cature by calling setSubmissionTime()
+    // may need to be changed **before** enqueuing
+    // as a result, currently MSubmitTime is later than MHostBaseTime
```

Comment:
Please add dots in the end of the sentences and start new sentences with with a capital letter